sympy/integrals/prde.py

```diff
@@ -359,35 +379,381 @@ def prde_no_cancel_b_small(b, Q, n, DE):
+
+    # Solutions of the original equation are
+    #    y = Sum(dj*fj, (j, 1, r)) + Sum(ei*hi, (i, 1, m)),
+    # where  ei == ci  (i = 1, ..., m),  when
+    # A*Matrix([c1, ..., cm]) == 0 and
+    # B*Matrix([c1, ..., cm, d1, ..., dr]) == 0
+
+    # Build combined constraint matrix with m + r + m columns.
+
+    r = len(f)
+    I = eye(m)
+    A = A.row_join(zeros(A.rows, r + m))
+    B = B.row_join(zeros(B.rows, m))
+    C = I.row_join(zeros(m, r)).row_join(-I)
+
+    return f + H, A.col_join(B).col_join(C)
+
+
+def prde_cancel_primitive(b, Q, n, DE):
+    """
+    Pg, 237
+    """
+    with DecrementLevel(DE):
+        ba, bd = frac_in(b, DE.t)
+        Qyn = [frac_in(q.nth(n), DE.t, field=True) for q in Q]
+        # fn = f1n, ..., frn_n
+        # An is a matrix with coefficients in Const(k)
+        # yn = Sum(djn*fjn, (j, 1, rn))
+        fn, An = param_rischDE(ba, bd, Qyn, DE)
+
+    fn = [Poly(fa.as_expr()/fd.as_expr(), DE.t, field=True)
+            for fa, fd in fn]
+
+    rn = len(fn)
+    m = len(An) - rn
+
+    # there would be a need to form a new equation
+    # just like Dy + fy = Sum(ci*qi, (i, 1, m))
+    # the equation is
+    # Dh + bh = Sum(ci*qi, (i, 1, m)) - Sum(djn*D(fjn*t**n) - b*fj*t**n)
+    # the below thing is wrong
+    for N in range(n, -1, -1):
+        s1 = 0
+        for j in range(rn):
+            djn = An[m + j]
+            fjn = fn[j]
+            s1 += djn*(derivation(fjn*DE.t**n, DE) - b*fjn*DE.t**n)
+
+
+def prde_cancel_exp(b, Q, n, DE):
+    """
+    Pg, 237
+    """
+    with DecrementLevel(DE):
+        # equation (7.23)
+        # Note: b != ba/bd
+        ba, bd = frac_in(b + n*derivation(DE.t, DE)/DE.t, DE.t)
+        Qyn = [(Poly(q.nth(n), DE.t), Poly(1, DE.t)) for q in Q]
+        fn, An = param_rischDE(ba, bd, Qyn, DE)
+
+    fn = [Poly(fa.as_expr()/fd.as_expr(), DE.t, field=True)
+            for fa, fd in fn]
+
+
+def param_poly_rischDE(a, b, q, n, DE):
+    """Polynomial solutions of a parametric Risch differential equation.
+
+    Given a derivation D in k[t], a, b in k[t] relatively prime, and q
+    = [q1, ..., qm] in k[t]^m, return h = [h1, ..., hr] in k[t]^r and
+    a matrix A with m + r columns and entries in Const(k) such that
+    a*Dp + b*p = Sum(ci*qi, (i, 1, m)) has a solution p of degree <= n
+    in k[t] with c1, ..., cm in Const(k) if and only if p = Sum(dj*hj,
+    (j, 1, r)) where d1, ..., dr are in Const(k) and (c1, ..., cm,
+    d1, ..., dr) is a solution of Ax == 0.
+    """
+    m = len(q)
+    if n < 0:
+        # Only the trivial zero solution is possible.
+        # Find relations between the qi.
+        if all([qi.is_zero for qi in q]):
+            return [], zeros(1, m)  # No constraints.
+
+        N = max([qi.degree(DE.t) for qi in q])
+        M = Matrix(N + 1, m, lambda i, j: q[j].nth(i))
+        A, _ = constant_system(M, zeros(M.rows, 1), DE)
+
+        return [], A
+
+    if a.is_ground:
+        # Normalization: a = 1.
+        a = a.LC()
```

Comment:
This one https://github.com/sympy/sympy/pull/11761/files#diff-25c3a80e410bd47d98bfac9be56b42b1R555 i.e `prde_spde`.
```python
[gxyd@stallman sympy{param-rde}]$ python3 ./bin/test sympy/integrals/tests/test_prde.py
============================================================= test process starts =============================================================
executable:         /home/gxyd/anaconda3/bin/python3  (3.6.0-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
random seed:        66232951
hash randomization: on (PYTHONHASHSEED=2396363264)

sympy/integrals/tests/test_prde.py[15] ..............E                                                                                   [FAIL]

_______________________________________________________________________________________________________________________________________________
_____________________________________________ sympy/integrals/tests/test_prde.py:test_issue_10798 _____________________________________________
  File "/home/gxyd/foss/sympy/sympy/integrals/tests/test_prde.py", line 278, in test_issue_10798
    assert integrate(1/(1-(x*y)**2), (x, 0, 1), y) == \
  File "/home/gxyd/foss/sympy/sympy/integrals/integrals.py", line 1295, in integrate
    risch=risch, manual=manual)
  File "/home/gxyd/foss/sympy/sympy/integrals/integrals.py", line 486, in doit
    conds=conds)
  File "/home/gxyd/foss/sympy/sympy/integrals/integrals.py", line 780, in _eval_integral
    result, i = risch_integrate(f, x, separate_integral=True, conds=conds)
  File "/home/gxyd/foss/sympy/sympy/integrals/risch.py", line 1692, in risch_integrate
    ans, i, b = integrate_primitive(fa, fd, DE)
  File "/home/gxyd/foss/sympy/sympy/integrals/risch.py", line 1339, in integrate_primitive
    q, i, b = integrate_primitive_polynomial(p, DE)
  File "/home/gxyd/foss/sympy/sympy/integrals/risch.py", line 1291, in integrate_primitive_polynomial
    rv = limited_integrate(aa, ad, [(Dta, Dtb)], DE)
  File "/home/gxyd/foss/sympy/sympy/integrals/prde.py", line 816, in limited_integrate
    h, A = param_rischDE(Fa, Fd, G, DE)
  File "/home/gxyd/foss/sympy/sympy/integrals/prde.py", line 716, in param_rischDE
    h, B = param_poly_rischDE(a, b, r, n, DE)
  File "/home/gxyd/foss/sympy/sympy/integrals/prde.py", line 555, in param_poly_rischDE
    a, b, q, r, n = prde_spde(a, b, q, n, DE)
  File "/home/gxyd/foss/sympy/sympy/integrals/prde.py", line 306, in prde_spde
    n1 = n - a.degree(DE.t)
AttributeError: 'One' object has no attribute 'degree'
```